### PR TITLE
add Group/Color, fix nix builds

### PR DESCRIPTION
**Pull request by tippenein**
I'm putting this out there for comments. I've been using it as-is with a nix build, but the stack machinery should still work

This also adds:
- Colors
- Groups 
- changes examples to use the lens'

### CODE REVIEW

**Comment by mbeidler** [comment]
I'll take a look. Why do we need the new `toJSVal_aeson` when that is already provided by **ghcjs-base**?

File: src/C3.hs [hunk 1]
```diff
@@ -8,7 +8,7 @@ module C3
 
 import Control.Monad ((<=<))
 #if __GLASGOW_HASKELL__ >= 800
-import JavaScript.JSON.Types.ToJSVal (toJSVal_aeson)
+import JSON.Types.ToJSVal (toJSVal_aeson)
```
Comment (mbeidler) [inline src/C3.hs]: I'm curious about why this is needed. GHCJS 8 still has `toJSVal_aeson` in `GHCJS.Marhal`, right?

File: app/Main.hs [hunk 2]
```diff
@@ -14,25 +15,25 @@ main = do
   js_createChartContainer "gauge-chart"
   js_createChartContainer "pie-chart"
   js_createChartContainer "chart"
-  gaugeChart <- generate gaugeChartOpt
-  pieChart <- generate pieChartOpt
+  gaugeChart <- generate $ def & bindTo .~ "gauge-chart"
```
Comment (mbeidler) [inline app/Main.hs]: `Control.Lens` is well suited for configuration APIs. Nice addition.

File: src/C3.hs (hunk 1 above)
Comment (tippenein) [inline src/C3.hs]:
```
 src/C3.hs:10:23: error:
    Module ‘GHCJS.Marshal’ does not export ‘toJSVal_aeson’
```
This was the error by the way. (in nix build)

**Comment by tippenein** [comment]
I'd like to fix the build oddities today, and if I can't we can merge all the other non-related pieces?

**Comment by mbeidler** [comment]
@tippenein When compiling with the latest GHCJS stack snapshot (GHC 8), I get two errors:

```
    • Could not deduce (PToJSVal Data.Text.Internal.Text)
        arising from a use of ‘pToJSVal’
      from the context: AE.ToJSON a
        bound by the type signature for:
                   toJSVal_aeson :: AE.ToJSON a => a -> IO JSVal
        at src/JSON/Types/ToJSVal.hs:15:1-45
    • In the first argument of ‘return’, namely ‘(pToJSVal t)’
      In the expression: return (pToJSVal t)
      In an equation for ‘convertValue’:
          convertValue (AE.String t) = return (pToJSVal t)
```

```
    • Could not deduce (ToJSVal Double) arising from a use of ‘toJSVal’
      from the context: AE.ToJSON a
        bound by the type signature for:
                   toJSVal_aeson :: AE.ToJSON a => a -> IO JSVal
        at src/JSON/Types/ToJSVal.hs:15:1-45
    • In the expression: toJSVal (realToFrac n :: Double)
      In an equation for ‘convertValue’:
          convertValue (AE.Number n) = toJSVal (realToFrac n :: Double)
      In an equation for ‘toJSVal_aeson’:
          toJSVal_aeson x
            = cv (AE.toJSON x)
            where
                cv = convertValue
                convertValue :: AE.Value -> IO JSVal
                convertValue AE.Null = return jsNull
                convertValue (AE.String t) = return (pToJSVal t)
                convertValue (AE.Array a)
                  = (\ (AI.SomeJSArray y) -> y)
                    <$> (AI.fromListIO =<< mapM convertValue (V.toList a))
                convertValue (AE.Number n) = toJSVal (realToFrac n :: Double)
                convertValue (AE.Bool b) = return (toJSBool b)
                convertValue (AE.Object o)
                  = do { obj@(OI.Object obj') <- OI.create;
                         .... }
```
I fixed the build by deleting the JSON folder and removing the `#CPP` dependent code.

I think this is not a nix-specific issue, but the specific nix environment used by `reflex`, no? My guess is that reflex is using an old build or a fork of ghcjs-base. `toJSVal_aeson` is in `GHCJS.Marshal` in the current version of `ghcjs-base`.

Are there nix users who don't want to use the reflex-platform nix configuration? Is there a solution to fixing the build that doesn't involve breaking it for stack?

**Comment by tippenein** [comment]
You're most likely right. Can you push that fix to master and I can fix the nix build on my end?

**Comment by mbeidler** [comment]
@tippenein I pushed the fix to master.

**Pull request merged**